scripts/docker_setup.sh

```diff
@@ -98,23 +100,74 @@ function move_ifaces() {
 			sudo ip netns exec pause ethtool -N "${ifaces[$i]}" flow-type tcp4 action 0
 			sudo ip netns exec pause ethtool -u "${ifaces[$i]}"
 		fi
+		if [ "$mode" == 'cndp' ]; then
+			# num queues
+                        num_q=1
+                        # start queue index
+                        start_q_idx=22
+                        # RSS using TC filter
+			setup_tc "${ifaces[$i]}" $num_q $start_q_idx
+		fi
 	done
 	setup_addrs
 }
 
+# Setup TC
+# Parameters: $1 = interface, $2 = number of queues, $3 = start queue index
+function setup_tc() {
+        # Interface name
+        iface=$1
+        # Number of queues
+        num_q=$2
+        # Start queue index
+        sq_idx=$3
+        sudo ip netns exec pause ethtool --offload $iface hw-tc-offload on
+        # Create two traffic control groups for the two queue sets - set 0 and set 1.
+        # queue set 1 will be used for dataplane traffic.
+        # queue set 0 will handle rest of the traffic (eg: control plane traffic).
+        # for e.g., 22@0 means 22 queues starting from queue id 0. 
+	# 4@22 mean 4 queues starting from queue id 22.
+        sudo ip netns exec pause tc qdisc add dev $iface root mqprio \
+		num_tc 2 map 0 1 queues $sq_idx@0 $num_q@$sq_idx hw 1 mode channel
+        sudo ip netns exec pause tc qdisc add dev $iface clsact
+}
+
+# Add TC rules for N3/N6/N9 access and core interface.
+# Parameters: $1 = access interface, $2 = core interface
+# Inner UE IP address range and GTPU port is hardcoded for now.
+# UE IP address should match the generated traffic pattern.
+function add_tc_rules() {
+        # Encapuslated traffic N3 on access interface.
+	# RSS GTPU filter (Note: hw_tc 1 has >1 queues which results in implict RSS)
+        sudo ip netns exec pause tc filter add dev $1 protocol ip ingress \
+		prio 1 flower src_ip 16.0.0.0/16 enc_dst_port 2152 skip_sw hw_tc 1
+        # List TC rules on access interface.
+        sudo ip netns exec pause tc filter show dev $1 ingress
+
+        # Encapsulated traffic N9 on core interface.
+	# RSS GTPU filter (Note: hw_tc 1 has >1 queues which results in implict RSS)
+        sudo ip netns exec pause tc filter add dev $2 protocol ip ingress \
+		prio 1 flower dst_ip 16.0.0.0/16 enc_dst_port 2152 skip_sw hw_tc 1
+        # un-encapsulated traffic N6 on core interface
+        sudo ip netns exec pause tc filter add dev $2 protocol ip ingress \
+		prio 1 flower dst_ip 16.0.0.0/16 skip_sw hw_tc 1
+        # List TC rules on core interface.
+        sudo ip netns exec pause tc filter show dev $2 ingress
+}
+
 # Stop previous instances of bess* before restarting
 docker stop pause bess bess-routectl bess-web bess-pfcpiface || true
 docker rm -f pause bess bess-routectl bess-web bess-pfcpiface || true
 sudo rm -rf /var/run/netns/pause
 
 # Build
-make docker-build
+#make docker-build
```

Comment:
Why is this change needed?

Reply:
Done

Reply:
Does commenting it out only helps when someone has already built the docker image, correct? In other words, it is needed for at least the first time the script is executed, correct?

Reply:
Yes. User can build the docker image using make docker-build first and then run the script.